src/elements/content-sidebar/__tests__/SidebarNavSign.test.tsx

```diff
@@ -1,80 +1,75 @@
+    const requestSignatureButtonText = localize(messages.boxSignRequestSignature.id);
+    const signMyselfButtonText = localize(messages.boxSignSignMyself.id);
 
     const renderComponent = (props = {}, features = {}) =>
         render(
             <FeatureProvider features={features}>
                 <SidebarNavSign {...props} />
             </FeatureProvider>,
+            {
+                wrapper: ({ children }: { children?: React.ReactNode }) => (
+                    <IntlProvider locale="en-US">{children}</IntlProvider>
+                ),
+            },
         );
 
-    test.each([true, false])('should render sign button', isRemoveInterstitialEnabled => {
-        const features = {
-            boxSign: {
-                isSignRemoveInterstitialEnabled: isRemoveInterstitialEnabled,
-            },
-        };
+    test('should render sign button', () => {
+        const wrapper = renderComponent();
 
-        const wrapper = renderComponent({}, features);
-        expect(wrapper.getByTestId('sign-button')).toBeVisible();
+        expect(wrapper.getByLabelText(signLabel)).toBeVisible();
     });
 
-    test('should call correct handler when sign button is clicked', () => {
-        const features = {
-            boxSign: {
-                isSignRemoveInterstitialEnabled: false,
-                onClick: onClickRequestSignature,
-            },
-        };
-        const { getByTestId } = renderComponent({}, features);
-
-        fireEvent.click(getByTestId('sign-button'));
+    test('should open dropdown with 2 menu items when sign button is clicked', () => {
+        const wrapper = renderComponent();
 
-        expect(onClickRequestSignature).toBeCalled();
-    });
+        fireEvent.click(wrapper.getByLabelText(signLabel));
 
-    test('should open dropdown with 2 menu items when sign button is clicked', () => {
-        const features = {
-            boxSign: {
-                isSignRemoveInterstitialEnabled: true,
-            },
-        };
-        const { getByTestId } = renderComponent({}, features);
-        fireEvent.click(getByTestId('sign-button'));
-        expect(getByTestId('sign-request-signature-button')).toBeVisible();
-        expect(getByTestId('sign-sign-myself-button')).toBeVisible();
+        expect(wrapper.getByText(requestSignatureButtonText)).toBeVisible();
+        expect(wrapper.getByText(signMyselfButtonText)).toBeVisible();
     });
 
     test('should call correct handler when request signature option is clicked', () => {
         const features = {
             boxSign: {
-                isSignRemoveInterstitialEnabled: true,
                 onClick: onClickRequestSignature,
             },
         };
-        const { getByTestId } = renderComponent({}, features);
-        fireEvent.click(getByTestId('sign-button'));
-        fireEvent.click(getByTestId('sign-request-signature-button'));
+        const wrapper = renderComponent({}, features);
```

Comment:
nit: is it possible to use screen instead of calling methods on the return value of the render function?

https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen

Reply:
agree, let's update it 👍